cognite/client/credentials.py

```diff
@@ -23,6 +24,69 @@ class CredentialProvider(Protocol):
     def authorization_header(self) -> tuple[str, str]:
         raise NotImplementedError
 
+    @classmethod
+    def load(cls, config: dict | str) -> CredentialProvider:
+        """Create a CredentialProvider from a configuration dictionary.
+
+        Args:
+            config (dict | str): A dictionary or dictionary parsable string containing the configuration for the credential provider.
+                The dictionary must contain exactly one top level key, which is the type of the credential provider and must be one of the following strings:
+                "token", "client_credentials", "interactive", "device_code", "client_certificate".
+                The value of the key is a dictionary containing the configuration for the credential provider.
+
+        Returns:
+            CredentialProvider: Initialized credential provider of the specified type.
+
+        Examples:
+                Get a token credential provider:
+
+                >>> from cognite.client.credentials import CredentialProvider
+                >>> config = {"token": "my secret token"}
+                >>> credential_provider = CredentialProvider.load(config)
+
+                Get a client credential provider:
+
+                >>> from cognite.client.credentials import CredentialProvider
+                >>> import os
+                >>> config = {
+                ...     "client_credentials": {
+                ...         "client_id": "abcd",
+                ...         "client_secret": os.environ["OAUTH_CLIENT_SECRET"],
+                ...         "token_url": os.environ["TOKEN_URL"],
+                ...         "scopes": ["https://greenfield.cognitedata.com/.default"],
+                ...         # Any additional IDP-specific token args. e.g.
+                ...         "audience": "some-audience",
+                ...     }
+                ... }
+                >>> credential_provider = CredentialProvider.load(config)
+        """
+
+        loaded = load_dict_or_str(config)
+
+        if len(loaded) != 1:
+            raise ValueError(
+                "Credential provider configuration must be a dictionary containing exactly one top level key."
+            )
+
+        credential_type, credential_config = next(iter(loaded.items()))
```

Comment:
If you want this to be less readable (🥇 ):
```suggestion
        (credential_type, credential_config), = loaded.items()
```

Reply:
the linters configured in my other projects would get so mad at that 🤣